subsys/bluetooth/host/hci_core.c

```diff
@@ -8048,6 +8201,18 @@ int bt_le_ext_adv_update_param(struct bt_le_ext_adv *adv,
 		return -EINVAL;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_PER_ADV) &&
+	    atomic_test_bit(adv->flags, BT_PER_ADV_PARAMS_SET)) {
+		/* If params for per adv has been set, do not allow setting
+		 * connectable, scanable or use legacy adv
+		 */
+		if (param->options & BT_LE_ADV_OPT_CONNECTABLE ||
+		    param->options & BT_LE_ADV_OPT_SCANNABLE ||
+		    param->options & BT_LE_ADV_OPT_EXT_ADV) {
```

Comment:
this should be `ANONYMOUS` instead

Reply:
What do you mean by "instead"? It was definitely missing  `BT_LE_ADV_OPT_ANONYMOUS` (thanks!), but the other 3 should still be present.

Reply:
It should be !(param->options & BT_LE_ADV_OPT_EXT_ADV), 
```
/* If params for per adv has been set, do not allow setting
 * connectable, scanable or use legacy adv
 */
```

Reply:
Nice catch